lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1457,255 +1348,127 @@ def test_lazy_data_masked__mask_set(self):
-        fill_value = 999
-        cube = self._sample_cube(dtype=np.int16, masked=True, lazy=True,
-                                 cube_fill_value=fill_value)
-        self.assertIs(cube.core_data().dtype, np.dtype('f8'))
+    def test_lazydata_realise(self):
+        # Check touching lazy data.
+        fill_value = 27.3
+        cube = self._sample_cube(lazy=True)
         data = cube.data
-        self.assertIs(cube.core_data().dtype, np.dtype('i2'))
-        self.assertEqual(cube.fill_value.dtype, np.dtype('i2'))
-        self.assertArrayAllClose(cube.fill_value, fill_value)
-        self.assertArrayAllClose(data.fill_value, fill_value)
+        self.assertIs(cube.core_data(), data)
+        self.assertEqual(cube.dtype, np.float32)
 
-    def test_lazydata_change(self):
-        # Check that re-assigning lazy data resets dtype and fill_value.
-        cube = self._sample_cube(dtype=np.float32, lazy=True,
-                                 cube_dtype=np.int16,
-                                 cube_fill_value=27.3)
-        # Set a modified dtype to check it gets reset.
-        self.assertEqual(cube.dtype, np.dtype('i2'))
-        self.assertEqual(cube.core_data().dtype, np.dtype('f4'))
-        self.assertArrayAllClose(cube.fill_value, 27)
-        new_data = self._sample_data(np.float64, lazy=True)
-        cube.data = new_data
-        self.assertEqual(cube.dtype, np.dtype('f8'))
-        self.assertIsNone(cube.fill_value)
-
-    def test_lazydata_nonmaskedints(self):
-        # Check that lazy ints retain their dtype.
-        data_original = self._sample_data(dtype=np.int32)
-        data_lazy = as_lazy_data(data_original)
-        cube = Cube(data_lazy)
-        self.assertEqual(cube.dtype, np.dtype('i4'))
-        self.assertEqual(cube.core_data().dtype, np.dtype('i4'))
-        data = cube.data
-        self.assertArrayEqual(data, data_original)
-
-    def test_lazydata_maskedints(self):
-        # Check lazy masked ints dtype.
-        fill_value = 1234
-        masked_data_original = self._sample_data(dtype=np.int32, masked=True)
-        masked_data_lazy = as_lazy_data(masked_data_original)
-        cube = Cube(masked_data_lazy, dtype=masked_data_original.dtype,
-                    fill_value=fill_value)
-        self.assertEqual(cube.dtype, np.dtype('i4'))
-        self.assertEqual(cube.core_data().dtype, np.dtype('f8'))
+    def test_lazymaskdata_realise(self):
+        # Check touching masked lazy data.
+        fill_value = 27.3
+        cube = self._sample_cube(masked=True, fill_value=fill_value, lazy=True)
         data = cube.data
-        self.assertTrue(ma.is_masked(data))
-        self.assertMaskedArrayEqual(data, masked_data_original)
-        self.assertArrayAllClose(cube.fill_value, fill_value)
+        self.assertIs(cube.core_data(), data)
+        self.assertEqual(cube.dtype, np.float32)
         self.assertArrayAllClose(data.fill_value, fill_value)
```

Comment:
I believe I may have written this, if not I think I certainly "could have done".
I interpret it as just a way of doing a toleranced floating-point comparison
 \- because it is IMHO  "basically always wrong" to test floating-point values for exact equivalance.